File: assets/js/videoChannel.js
import socket from "./socket";

let Joinedchannel;
const joinButton = document.getElementById("join-channel-btn");
const channelInput = document.getElementById("channel-input-name");
const pushButton = document.getElementById("push-message-btn");
const messageInput = document.getElementById("message-input-field");
const nameButton = document.getElementById("name-btn");
const nameInput = document.getElementById("input-name");
let name;
nameButton.addEventListener("click", function (e) {
    name = nameInput.value;
})

joinButton.addEventListener("click", function (event) {
    let channelName = channelInput.value;
    Joinedchannel = socket.channel("video:" + channelName, { "name": name });
    Joinedchannel.on("ping",
        ({ count, topic }) => console.log(`PING from ${topic}::`, count))
    Joinedchannel.on("pong",
        ({ count, topic }) => console.log(`PONG from ${topic}::`, count))
    Joinedchannel.on("new_message", ({ user }) => console.log(`username ${user.username}:: message ${user.message}`))

    pushButton.addEventListener("click", function (event) {
        let message = messageInput.value;
        Joinedchannel.push("new_message", { message: message })
            .receive("error", (e) => {
                console.log(e)
            })
        messageInput.value = "";

    })

    Joinedchannel.join()
        .receive("ok", resp => { console.log("Joined successfully", resp) })
        .receive("error", resp => { console.log("Unable to join", resp) })
});

File: lib/realtimeapp_web/channels/video_channel.ex
defmodule RealtimeappWeb.VideoChannel do
  use RealtimeappWeb, :channel

  def join("video:" <> _video_id, params, socket) do
    # :timer.send_interval(5_000, :ping)
    # :timer.send_interval(1_000, :pong)
    # topic is in socket by default
    IO.inspect(params)
    {:ok, assign(socket, :username, params["name"])}
  end

  def handle_info(:ping, socket) do
    count = socket.assigns[:count] || 1
    username = socket.assigns[:username] || nil
    push(socket, "ping", %{count: count + 1, username: username})
    {:noreply, assign(socket, :count, count + 1)}
  end

  def handle_info(:pong, socket) do
    count = socket.assigns[:count] || 1
    username = socket.assigns[:topiusernamec] || nil
    push(socket, "pong", %{count: count + 1, username: username})
    {:noreply, assign(socket, :count, count + 1)}
  end

  def handle_in("new_message", payload, socket) do
    broadcast(socket, "new_message", %{
      user: %{username: socket.assigns[:username], message: payload["message"]}
    })

    {:reply, :ok, socket}
  end
end
